pkg/printer/printer.go

```diff
@@ -194,65 +194,123 @@ func (p *Printer) WriteProtoFlow(res *observerpb.GetFlowsResponse) error {
 	f := res.GetFlow()
 	switch p.opts.output {
 	case TabOutput:
+		var err error
+		src, dst := p.GetHostNames(f)
+
 		if p.line == 0 {
-			_, err := fmt.Fprint(p.tw,
-				"TIMESTAMP", tab,
-				"SOURCE", tab,
-				"DESTINATION", tab,
-				"TYPE", tab,
-				"VERDICT", tab,
-				"SUMMARY", newline,
-			)
-			if err != nil {
-				return err
+			if !p.opts.nodeName {
+				_, err = fmt.Fprint(p.tw,
+					"TIMESTAMP", tab,
+					"SOURCE", tab,
+					"DESTINATION", tab,
+					"TYPE", tab,
+					"VERDICT", tab,
+					"SUMMARY", newline,
+				)
+				if err != nil {
+					return err
+				}
+			} else {
+				_, err = fmt.Fprint(p.tw,
+					"TIMESTAMP", tab,
+					"NODE", tab,
+					"SOURCE", tab,
+					"DESTINATION", tab,
+					"TYPE", tab,
+					"VERDICT", tab,
+					"SUMMARY", newline,
+				)
+				if err != nil {
+					return err
+				}
 			}
 		}
-		src, dst := p.GetHostNames(f)
-		_, err := fmt.Fprint(p.tw,
-			fmtTimestamp(f.GetTime()), tab,
-			src, tab,
-			dst, tab,
-			GetFlowType(f), tab,
-			f.GetVerdict().String(), tab,
-			f.GetSummary(), newline,
-		)
+		if !p.opts.nodeName {
+			_, err = fmt.Fprint(p.tw,
+				fmtTimestamp(f.GetTime()), tab,
+				src, tab,
+				dst, tab,
+				GetFlowType(f), tab,
+				f.GetVerdict().String(), tab,
+				f.GetSummary(), newline,
+			)
+		} else {
+			_, err = fmt.Fprint(p.tw,
+				fmtTimestamp(f.GetTime()), tab,
+				f.GetNodeName(), tab,
+				src, tab,
+				dst, tab,
+				GetFlowType(f), tab,
+				f.GetVerdict().String(), tab,
+				f.GetSummary(), newline,
+			)
+		}
```

Comment:
I'm not really fond of all this duplication just to add a new field when printing. This could lead to unwanted discrepancies based on whether the node name flag is active or not. What about something like this instead?
```go
func (p *Printer) WriteProtoFlow(res *observerpb.GetFlowsResponse) error {
        var err error
        write := func(w io.Writer, a ...interface{}) {
                if err != nil {
                        return
                }
                _, err = fmt.Fprint(w, a...)
        }

        f := res.GetFlow()
        switch p.opts.output {
        case TabOutput:
                var err error
                src, dst := p.GetHostNames(f)
                if p.line == 0 {
                        write(p.tw, "TIMESTAMP", tab)
                        if p.opts.nodeName {
                                write(p.tw, "NODE", tab)
                        }
                        write(p.tw,
                                "SOURCE", tab,
                                "DESTINATION", tab,
                                "TYPE", tab,
                                "VERDICT", tab,
                                "SUMMARY", newline,
                        )
                }
                write(p.tw, fmtTimestamp(f.GetTime()), tab)
                if p.opts.nodeName {
                        write(p.tw, f.GetNodeName(), tab)
                }
                write(p.tw,
                        src, tab,
                        dst, tab,
                        GetFlowType(f), tab,
                        f.GetVerdict().String(), tab,
                        f.GetSummary(), newline,
                )
                if err != nil {
                        return fmt.Errorf("failed to write out packet: %v", err)
                }
        case DictOutput:
            // apply the same method as for TabOutput, etc
            ...
```